dcos/cosmos.py

```diff
@@ -180,7 +180,8 @@ def _cosmos_request(self,
                         response.headers.get('Content-Type')))
             return response
         except DCOSBadRequest as e:
-            if len(headers_preference) > 1:
+            error_type = e.response.json().get('type')
+            if error_type == 'not_valid' and len(headers_preference) > 1:
```

Comment:
@larioj I think we should fix this in Cosmos first. I want our error types to be consistent. This is not consistent with the rest of our system.